### PR TITLE
pixel snapping

### PR DESCRIPTION
**Pull request by Planutim**
divided `modelviews` uniform to `model` and `view` for easier pixel rounding calculation

### CODE REVIEW

File: game/processinput.go [hunk 1]
```diff
@@ -41,13 +41,19 @@ func ProcessInput() {
...
 	case input.FREECAM:
 		if input.GetButtonDown("freecam-off") {
 			Cam.TurnOffFreeCam()
 		}
 		if input.GetButtonDown("toggle-bbox") {
 			render.ToggleBBox()
 		}
+		if input.GetButtonDown("toggle-pixel-snap") {
+			render.TogglePixelSnap()
+		}
```
Comment (ted537) [inline game/processinput.go:56]: Since there are multiple options, this should be named `CyclePixelSnap()` imo

File: engine/input/keyboard.go [hunk 2]
```diff
@@ -36,14 +41,22 @@ func keyCallback(
 	}
 
 	if action == glfw.Press {
+		if key == glfw.KeyU {
+			timer = time.Now()
+		}
```
Comment (ted537) [inline engine/input/keyboard.go:46]: Should this be called `debugBubbleTimer`?

File: render/sprite-draw.go [hunk 3]
```diff
@@ -221,3 +230,19 @@ func ToggleFiltering() {
...
+		fmt.Println("PIXEL SNAPPING: snapping to a pixel")
+	case &spriteProgram2:
+		spriteProgram = &spriteProgram3
+		fmt.Println("PIXEL SNAPPING: snapping to half a pixel")
+	case &spriteProgram3:
+		spriteProgram = &spriteProgram1
+		fmt.Println("PIXEL SNAPPING: no snapping")
+	default:
+		log.Fatalf("Pixel snapping error")
+	}
```
Comment (ted537) [inline render/sprite-draw.go:247]: Storing a list and an index may be simpler than switch/casing pointers

File: render/sprite.go [hunk 4]
```diff
@@ -8,25 +8,52 @@ import (
...
-	spriteProgram = config.Compile()
+	config2.Define("NUM_INSTANCES",
+		strconv.Itoa(int(constants.DRAW_SPRITE_BATCH_SIZE)))
+
+	config3.Define("NUM_INSTANCES",
+		strconv.Itoa(int(constants.DRAW_SPRITE_BATCH_SIZE)))
+
+	spriteProgram1 = config1.Compile()
+	spriteProgram2 = config2.Compile()
+	spriteProgram3 = config3.Compile()
```
Comment (ted537) [inline render/sprite.go:44]: We can probably do better naming than 1/2/3.

File: game/update.go [hunk 5]
```diff
@@ -24,7 +24,7 @@ func Update(dt float32) {
 		//playerPos := player.InterpolatedTransform.Col(3).Vec2()
 		playerPos :=
 			World.Entities.Agents.FromID(playerAgentID).PhysicsState.Pos
-		Cam.SetCameraPosition(playerPos.X, playerPos.Y)
+		Cam.SetCameraPosition(playerPos.X+player.PhysicsParameters.Size.X/2, playerPos.Y)
```
Comment (ted537) [inline game/update.go:27]: Isn't player's position already centered?

File: engine/input/keyboard.go [hunk 6]
```diff
@@ -36,14 +41,22 @@ func keyCallback(
...
 
 		}
 		lastKeyPressed = key
 		KeysPressedDown[key] = true
 		KeysPressed[key] = true
+
 	} else if action == glfw.Repeat {
+		if key == glfw.KeyU {
+			newTime := time.Since(timer).Milliseconds()
+			fmt.Println(newTime)
```
Comment (ted537) [inline engine/input/keyboard.go:58]: Should probably not print numbers without context to `stdout`

File: game/update.go (hunk 5 above)
Comment (Planutim) [reply game/update.go:27]: on bigger zoom levels it zooms left and top from the player, i wanted to offset to see player and tile pixels

File: engine/input/keyboard.go (hunk 6 above)
Comment (Planutim) [reply engine/input/keyboard.go:58]: this is debug didn't delete before pull requesting

File: render/sprite-draw.go (hunk 3 above)
Comment (Planutim) [reply render/sprite-draw.go:247]: will change it

**Pull request merged**